core/src/messages/dns.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{
+    nats::{SubscribeSubject, TypedMessage},
+    types::ClusterName,
+};
+use serde::{Deserialize, Serialize};
+use trust_dns_server::client::rr::RecordType;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum LocalDnsRecordType {
+    A,
+    TXT,
+    AAAA,
+    Unknown(u16),
+}
+
+impl LocalDnsRecordType {
+    fn as_trust_record_type(&self) -> RecordType {
+        match self {
+            LocalDnsRecordType::A => RecordType::A,
+            LocalDnsRecordType::TXT => RecordType::TXT,
+            LocalDnsRecordType::AAAA => RecordType::AAAA,
+            LocalDnsRecordType::Unknown(id) => RecordType::Unknown(*id),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SetDnsRecord {
+    pub cluster: ClusterName,
```

Comment:
I like that!

Reply:
I think this should be a `String`, because in the case of a `TXT` record its value is `_acme-challenge` which is not a `ClusterName`. We could create a "dummy" `ClusterName` for this but that would violate the newtype pattern.

Reply:
ah I completely misunderstood something. does the DNS server itself know what domain its running on?

Reply:
like the record request for acme is _acme_challenge.<cluster> right? where cluster is the domain associated with a cluster? hmmm I think im misunderstanding something very simple

Reply:
and it'll need a ClusterName for the nats subject (for TXT records)

Reply:
Ahh sorry ignore me ignore me

I was confusing name and cluster_name. We still need this, and you added name as a string, which is right.